cpp/src/arrow/dataset/partition.cc

```diff
@@ -61,6 +61,16 @@ Result<std::string> SafeUriUnescape(util::string_view encoded) {
   }
   return decoded;
 }
+
+std::string StripNonPrefix(const std::string& path) {
```

Comment:
Add a short comment describing this?

Reply:
Added a comment explaining the function.